design/repository-maintenance.md

```diff
@@ -0,0 +1,338 @@
+	checkError(err, log)
+	// Initialize repo provider
+	repoProvider := provider.NewUnifiedRepoProvider(
+		credentials.CredentialGetter{
+			FromFile:   credentialFileStore,
+			FromSecret: credentialSecretStore,
+		}, o.RepoType, cli, log)
+	
+	// Get backupRepository
+	repo, err := repository.GetBackupRepository(context.Background(), cli, f.Namespace(),
+		repository.BackupRepositoryKey{
+			VolumeNamespace: o.VolumeNamespace,
+			BackupLocation:  o.BackupStorageLocation,
+			RepositoryType:  o.RepoType,
+		}, true)
+	checkError(err, log)
+	// Get BSL
+	bsl := &velerov1api.BackupStorageLocation{}
+	cli.Get(context.Background(), client.ObjectKey{Namespace: f.Namespace(), Name: repo.Spec.BackupStorageLocation}, bsl)
+	checkError(err, log)
+
+	para := provider.RepoParam{
+		BackupRepo:     repo,
+		BackupLocation: bsl,
+	}
+	// Connect
+	err = repoProvider.BoostRepoConnect(context.Background(), para)
+	checkError(err, log)
+	// Prune repo
+	err = repoProvider.PruneRepo(context.Background(), para)
+	checkError(err, log)
+	...
+}
+```
+
+### 2. Create Jobs by Repository Manager
+Currently, the backup repository controller will call the repository manager to do the `PruneRepo`, and Kopia or Restic maintenance is then finally called through multiple chain calls.
+
+We will keep using the `PruneRepo` function in the repository manager, but we cut off the multiple chain calls by creating a maintenance job.
+
+The job definition would be like below:
+```yaml
+apiVersion: v1
+items:
+- apiVersion: batch/v1
+  kind: Job
+  metadata:
+    creationTimestamp: "2024-02-19T02:22:36Z"
+    generation: 1
+    # labels or affinity or topology settings would inherit from the velero deployment
+    labels:
+      batch.kubernetes.io/controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+      batch.kubernetes.io/job-name: nginx-example-default-kopia-pqz6c
+      controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+      job-name: nginx-example-default-kopia-pqz6c
+    name: nginx-example-default-kopia-pqz6c
+    namespace: velero
+  spec:
+    backoffLimit: 6
+    completionMode: NonIndexed
+    completions: 1
+    parallelism: 1
+    selector:
+      matchLabels:
+        batch.kubernetes.io/controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+    suspend: false
+    template:
+      metadata:
+        creationTimestamp: null
+        labels:
+          batch.kubernetes.io/controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+          batch.kubernetes.io/job-name: nginx-example-default-kopia-pqz6c
+          controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+          job-name: nginx-example-default-kopia-pqz6c
+        name: kopia-maintenance-job
+      spec:
+        containers:
+        # arguments for repo maintenance job
+        - args:
+          - server
+          - repo-maintenance
+          - --repo-name=nginx-example
+          - --repo-type=kopia
+          - --backup-storage-location=default
+          - --keep-latest-maintenance-jobs=3
```

Comment:
With `--keep-latest-maintenance-jobs`, is repo maintenance job responsible to clean up the job history?

Reply:
It's the `repository manager` response for maintenance jobs including creating, cleaning up jobs.
it's not cleaned by repo maintenance job